src/gf.c

```diff
@@ -1694,8 +1702,11 @@ jl_tupletype_t *jl_argtype_with_function(jl_function_t *f, jl_tupletype_t *types
     return (jl_tupletype_t*)tt;
 }
 
+#if 0
```

Comment:
maybe just add a comment: `TODO: revise this logic and re-enable. disabled during type-system rewrite.`